cpp/src/bitmask/null_mask.cu

```diff
@@ -371,37 +404,6 @@ cudf::size_type count_unset_bits(bitmask_type const* bitmask,
   return (num_bits - detail::count_set_bits(bitmask, start, stop, stream));
 }
 
-// Returns the bitwise AND of the null masks of all columns in the table view
-rmm::device_buffer bitmask_and(table_view const& view,
-                               rmm::cuda_stream_view stream,
-                               rmm::mr::device_memory_resource* mr)
-{
-  CUDF_FUNC_RANGE();
-  rmm::device_buffer null_mask{0, stream, mr};
-  if (view.num_rows() == 0 or view.num_columns() == 0) { return null_mask; }
-
-  std::vector<bitmask_type const*> masks;
-  std::vector<size_type> offsets;
-  for (auto&& col : view) {
-    if (col.nullable()) {
-      masks.push_back(col.null_mask());
-      offsets.push_back(col.offset());
-    }
-  }
-
-  if (masks.size() > 0) {
-    return cudf::detail::bitmask_binop(
-      [] __device__(bitmask_type left, bitmask_type right) { return left & right; },
-      masks,
-      offsets,
-      view.num_rows(),
-      stream,
-      mr);
-  }
-
-  return null_mask;
-}
-
 // Returns the bitwise OR of the null masks of all columns in the table view
 rmm::device_buffer bitmask_or(table_view const& view,
```

Comment:
Is a separate PR planned to do this same refactoring for `bitmask_or`? The return types of `bitmask_and` and `bitmask_or` should be the same.

Reply:
Good point, the initial goal to make `bitmask_and` return valid count was to help hash join optimization and during this work, I realized that other implementations would benefit from it as well. I haven't looked at `bitmask_or` in detail, but I assume similar improvement can be applied to many `bitmask_or` use cases also. I will create a separate PR to update `bitmask_or` if that's the case. Thanks for the reminder.

Reply:
**Updates**: except for unit tests, looks like there are no real use cases of `bitmask_or`.

Reply:
Since the scope is very limited, I would recommend changing its return type in this PR.